App/Controllers/AuthController.php

```diff
@@ -17,30 +23,38 @@ public function login()
         return $this->render('auth/login');
     }
 
-    public function auth(Request $request)
+    /**
+     * auth
+     *
+     * @param  mixed $request
+     * @return void
+     */
+    public function auth(HttpRequest $request)
     {
-        if ($request->getMethod() === 'POST') {
-            $username = $request->username;
-            $password = $request->password;
+        $username = $request->get('username');
+        $password = $request->get('password');
 
-            $user = User::where('username', $request->username);
+        $users = User::where('username', $username);
+        $user = array_shift($users);
 
-            if (!empty($user)) {
-                if (password_verify($password, $user['password'])) {
-                    Session::set('logged_in', true);
-                    Session::set('username', $username);
-                    redirect('/news');
-                }
-            } else {
-                return $this->render('auth/login', [
-                    'error' => 'Wrong Login Data!'
-                ]);
-            }
+        if (!empty($user) && password_verify($password, $user->password)) {
+            Session::set('logged_in', true);
+            Session::set('username', $username);
+            redirect('/news');
+        } else {
+            Session::set('error', 'Wrong Login Data!');
+            redirect('/');
         }
     }
```

Comment:
```suggestion
        if (empty($user) || !password_verify($password, $user->password)) {
            Session::set('error', 'Wrong Login Data!');
            redirect('/');
            
            return;
        }

        Session::set('logged_in', true);
        Session::set('username', $username);
        redirect('/news');
    }
```
Generally speaking, you filter out everything that can go wrong first. The last bit of the function should be the happy flow